aspnetcore/security/authentication/claims.md

```diff
@@ -27,121 +27,38 @@ This article covers the following areas:
-        options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
-    })
-   .AddCookie()
-   .AddOpenIdConnect(options =>
-   {
-       options.SignInScheme = "Cookies";
-       options.Authority = "-your-identity-provider-";
-       options.RequireHttpsMetadata = true;
-       options.ClientId = "-your-clientid-";
-       options.ClientSecret = "-your-client-secret-from-user-secrets-or-keyvault";
-       options.ResponseType = "code";
-       options.UsePkce = true;
-       options.Scope.Add("profile");
-       options.SaveTokens = true;
-       options.GetClaimsFromUserInfoEndpoint = true;
-       options.ClaimActions.MapUniqueJsonKey("preferred_username", "preferred_username");
-       options.ClaimActions.MapUniqueJsonKey("gender", "gender");
-   }); 
-```
+Another way to get the user claims is to use the OpenID Connect User Info API. The ASP.NET Core client app uses the `GetClaimsFromUserInfoEndpoint` property to configure this. One important difference from the first settings, is that you must specify the claims you require using the `MapUniqueJsonKey` method, otherwise only the `name`, `given_name` and `email` standard claims will be available in the client app. The claims included in the `id_token` are mapped per default. This is the major difference to the first option. You must explicitly define some of the claims you require.
+
+[!code-csharp[](~/security/authentication/claims/sample6/WebRPmapClaims/Program.cs?name=snippet2&highlight=10-31)]
 
 ## Name claim and role claim mapping
 
 The **Name** claim and the **Role** claim are mapped to default properties in the ASP.NET Core HTTP context. Sometimes it is required to use different claims for the default properties, or the name claim and the role claim do not match the default values. The claims can be mapped using the **TokenValidationParameters** property and set to any claim as required. The values from the claims can be used directly in the HttpContext **User.Identity.Name** property and the roles.
 
 If the `User.Identity.Name` has no value or the roles are missing, please check the values in the returned claims and set the `NameClaimType` and the `RoleClaimType` values. The returned claims from the client authentication can be viewed in the HTTP context.
 
-```csharp
-public void ConfigureServices(IServiceCollection services)
-{
-    services.AddAuthentication(options =>
-    {
-        options.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
-        options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
-    })
-   .AddCookie()
-   .AddOpenIdConnect(options =>
-   {
-       // other options...
-       options.TokenValidationParameters = new TokenValidationParameters
-       {
-         NameClaimType = "email", 
-         // RoleClaimType = "role"
-       };
-   });
-```
+[!code-csharp[](~/security/authentication/claims/sample6/WebRPmapClaims/Program.cs?name=snippet_name&highlight=10-24)]
```

Comment:
I think this could also be shrunk down, and just the `TokenValidationParameters` assignment could be highlighted.